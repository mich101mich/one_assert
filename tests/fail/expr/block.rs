fn main() {
    one_assert::assert!({
        let x = 5;
        x + 1
    });
    one_assert::assert!({
        let x = 5;
        x == 5;
    });
    one_assert::assert!({
        let x = 5;
    });
}
