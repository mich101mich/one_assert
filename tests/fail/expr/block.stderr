error[E0308]: mismatched types
 --> tests/fail/expr/block.rs:4:9
  |
4 |         x + 1
  |         ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
 --> tests/fail/expr/block.rs:6:25
  |
6 |       one_assert::assert!({
  |  _________________________^
7 | |         let x = 5;
8 | |         x == 5;
  | |               - help: remove this semicolon to return this value
9 | |     });
  | |_____^ expected `bool`, found `()`

error[E0308]: mismatched types
  --> tests/fail/expr/block.rs:10:25
   |
10 |       one_assert::assert!({
   |  _________________________^
11 | |         let x = 5;
12 | |     });
   | |_____^ expected `bool`, found `()`
